app/crypto/key-encryption.ts

```diff
@@ -0,0 +1,51 @@
+import { Buffer } from 'buffer';
+
+const algorithmName = 'AES-CBC';
+
+function extractEncryptionKey(hash: Uint8Array) {
+  return hash.slice(0, 32);
+}
+
+function extractEncryptionInitVector(hash: Uint8Array) {
+  return hash.slice(32, hash.length);
+}
+
+async function deriveWebCryptoKey(derivedKeyHash: Uint8Array) {
+  const format = 'raw';
+  const key = extractEncryptionKey(derivedKeyHash);
+  const algorithm = { name: algorithmName };
+  const extractable = false;
+  const keyUsages: KeyUsage[] = ['encrypt', 'decrypt'];
+  return crypto.subtle.importKey(format, key, algorithm, extractable, keyUsages);
+}
+
+interface EncryptMnemonicArgs {
+  mnemonic: string;
+  derivedKeyHash: Uint8Array;
+}
+
+export async function encryptMnemonic({ mnemonic, derivedKeyHash }: EncryptMnemonicArgs) {
+  const key = await deriveWebCryptoKey(derivedKeyHash);
+  const iv = extractEncryptionInitVector(derivedKeyHash);
+  const cipherArrayBuffer = await crypto.subtle.encrypt(
+    { name: algorithmName, iv },
+    key,
+    new TextEncoder().encode(mnemonic)
+  );
+  return Buffer.from(cipherArrayBuffer).toString('hex');
+}
+
+interface DecryptMnemonicArgs {
+  encryptedMnemonic: string;
+  derivedKeyHash: Uint8Array;
+}
+
+export async function decryptMnemonic({ encryptedMnemonic, derivedKeyHash }: DecryptMnemonicArgs) {
+  const key = await deriveWebCryptoKey(derivedKeyHash);
+  const iv = extractEncryptionInitVector(derivedKeyHash);
+  const algorithm = { name: 'AES-CBC', iv };
+  const encryptedBuffer = new Buffer(encryptedMnemonic, 'hex');
+  const decrypted = await crypto.subtle.decrypt(algorithm, key, encryptedBuffer);
+  const textDecoder = new TextDecoder();
+  return textDecoder.decode(decrypted);
+}
```

Comment:
Could you add some TSDoc comments to a few of these functions? It'd help me understand what some of these arguments and functions are, and I'm sure would help other devs that read this, too.